src/include/OSL/oslnoise.h

```diff
@@ -199,22 +199,6 @@ reference_inthash (const unsigned int k[N]) {
     return c;
 }
 
-
-template<typename HeadT, typename... ListT>
-using RequireAllSame = typename std::enable_if<conjunction<std::is_same<ListT, HeadT>...>::value>::type;
-
-// Allow any number of arguments to be adapted to the array based reference_inthash
-// and leave door open for overloads of inthash with specific parameters.
-// NOTE: only accept unsigned int's so that overloads with unsigned int's will be chosen.
-// Callers with "int" parameters will need to static_cast<unsigned int>(param)
-template<typename ...ListT, typename = RequireAllSame<unsigned int, ListT...> >
-OSL_FORCEINLINE OSL_HOSTDEVICE unsigned int
-inthash (ListT... uintList)  {
-    const int count = sizeof...(uintList);
-	const unsigned int iv[count] = {uintList...};
-	return reference_inthash<count>(iv);
-}
-
 #if (__OSL_USE_REFERENCE_INT_HASH == 0)
```

Comment:
Sorry to burden you with this, Declan, but can/should we eliminate these guards and use of the `__OSL_USE_REFERENCE_INT_HASH` symbol?

Reply:
I'm just thinking, if this serves no purpose, then let's not make the next person to come through this section of code get confused need to hunt for what that's supposed to be doing and under what circumstances it might be nonzero.

Reply:
Hmm it could be worthwhile getting rid of the `__OSL_USE_REFERENCE_INT_HASH` given that if this is non zero then it will fail to compile anyway because it removes all the inthash functions. So at this point your going to have to do some big modifications to the header anyway. We would have to decide what to do with the second definition of the variadic `inthashVec` function though as these would clash.

Reply:
I'll merge this now, and then I can fix the definition in a second pass.

Reply:
I guess this is fine, the purpose of the inthash (ListT... uintList) was to be easily fall back to the reference version in case of code gen or correctness issues.  If we are comfortable/confident then such a fallback mechanism may not be necessary, still suggest keeping reference_hash around (maybe inside a comment) so readers could trace back what exactly the optimized versions are doing.